pkg/gofr/http/multipart_file_bind.go

```diff
@@ -146,13 +163,194 @@ func (uf *formData) setFieldValue(value reflect.Value, data string) (bool, error
+	} else if value.Kind() == reflect.Array {
+		if len(elements) > value.Len() {
+			return false, errDataLengthExceeds
+		}
+
+		newSlice = reflect.New(value.Type()).Elem() // Create a new zero-valued array
+	} else {
+		return false, fmt.Errorf("%w: %s", errUnsupportedKind, value.Kind())
+	}
+
+	// Create a reusable element value to avoid unnecessary allocations
+	elemValue := reflect.New(elemType).Elem()
+
+	// Set the elements of the slice/array
+	for i, strVal := range elements {
+		// Update the reusable element value
+		if _, err := uf.setFieldValue(elemValue, strVal); err != nil {
+			return false, fmt.Errorf("%w %d: %w", errSettingValue, i, err)
+		}
+
+		newSlice.Index(i).Set(elemValue)
+	}
+
+	value.Set(newSlice)
+
+	return true, nil
+}
+
+func (*formData) setStructValue(value reflect.Value, data string) (bool, error) {
+	if value.Kind() != reflect.Struct {
+		return false, errNotAStruct
+	}
+
+	dataMap, err := parseStringToMap(data)
+	if err != nil {
+		return false, err
+	}
+
+	anyFieldSet := false
+
+	for key, val := range dataMap {
+		field, err := getFieldByName(value, key)
+		if err != nil {
+			return false, err
+		}
+
+		if err := setFieldValueFromData(field, val); err != nil {
+			return false, err
+		}
+
+		anyFieldSet = true
+	}
+
+	return anyFieldSet, nil
+}
+
+// getFieldByName retrieves a field by its name, considering case insensitivity.
+func getFieldByName(value reflect.Value, key string) (reflect.Value, error) {
+	field := value.FieldByName(key)
+	if !field.IsValid() {
+		field = findFieldByNameIgnoreCase(value, key)
+		if !field.IsValid() {
+			return reflect.Value{}, fmt.Errorf("%w: %s", errFieldNotFound, key)
+		}
+	}
+
+	if !field.CanSet() {
+		return reflect.Value{}, fmt.Errorf("%w: %s", errFieldUnexported, key)
+	}
+
+	return field, nil
+}
+
+// setFieldValueFromData sets the field's value based on the provided data.
+func setFieldValueFromData(field reflect.Value, data interface{}) error {
+	switch val := data.(type) {
+	case string:
+		field.SetString(val)
+	case int:
+		field.SetInt(int64(val))
+	case float64:
+		field.SetFloat(val)
+	case bool:
+		field.SetBool(val)
 	default:
-		return false, nil
+		return fmt.Errorf("%w: %s, %T", errUnsupportedFieldType, field.Type().Name(), val)
+	}
+
+	return nil
+}
+
+type customUnmarshaller struct {
+	dataMap map[string]interface{}
+}
+
+// UnmarshalJSON is a custom unmarshaller because json package in Go unmarshal numbers to float64 by default.
+func (c *customUnmarshaller) UnmarshalJSON(data []byte) error {
+	var rawData map[string]interface{}
+
+	err := json.Unmarshal(data, &rawData)
+	if err != nil {
+		return err
 	}
+
+	dataMap := make(map[string]interface{}, len(rawData))
+
+	for key, val := range rawData {
+		switch val := val.(type) {
+		case float64:
+			if val == float64(int(val)) {
+				dataMap[key] = int(val)
+			} else {
+				dataMap[key] = val
+			}
+		default:
+			dataMap[key] = val
+		}
```

Comment:
This seems very complicated

What about this?

```suggestion
		if valFloat, ok := val.(float64); ok {
			valInt := int(val)
			if valFloat == float64(valInilt) {
				val = valInt
			}
		}
		dataMap[key] = val
```

Reply:
simplied it